src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentationMeterProviderBuilderExtensions.cs

```diff
@@ -24,25 +20,6 @@ public static MeterProviderBuilder AddAspNetCoreInstrumentation(
     {
         Guard.ThrowIfNull(builder);
 
-#if NET
```

Comment:
What if we did...

```csharp
#if NETSTANDARD2_0_OR_GREATER
   throw new NotSupportedException("Metrics instrumentation is not supported when targeting .NET Standard.");
#endif
```

Reply:
I am not super familiar with the resolving libraries by nuget, I assume that for all out-of-the-box supported scenarios we will use net8 library. This one can be loaded in net6.0 probably.

I do not like the idea of throwing exceptions, ecen during startup. On non-supported framework current implementation will work more or less like no-op.

What is the reason behind throwing exception on netstandar2.0?

Reply:
Let's say we release this new version. If users try to load it into something < net8.0 then they will bind to this netstandard2.0 DLL. They won't be redirected to the ne8.0 one (I don't think, but we could verify). Doing this they should get a warning (from DS package), but they may suppress that and do it anyway.

So the idea with this throw is that if users try to add AspNetCore metrics instrumentation on something < net8.0 they get an exception telling them it isn't supported. Without the exception the app will start running, but nothing will show up because the Meters being listened to are only there net8.0+ (IIRC).

Reply:
I have the same question. If the `else` block is removed, what would happen on NetStandard? Should we be dropping NetStandard in `TargetFrameworks` tags as well?

Reply:
@CodeBlanch, what do ou think about something like this?

```csharp
#if NETSTANDARD2_0_OR_GREATER
if (System.Environement.Version.Major < 8)
{
   throw new NotSupportedException("Metrics instrumentation is not supported when executing on .NET 7 and lower..");
}
#endif
```

Ref: https://learn.microsoft.com/en-us/dotnet/api/system.environment.version?view=net-8.0

@xiang17, there were strong demand in this whole repository to include netstandard2.0 in every package..Some users implements wrappers around OTel with netsdanrd2.0 libs. As I remember, they are referencing them both in net and .net fx.

Reply:
@Kielek I'm good with that! FYI you have double-periods in the exception message ".." 😄 Could also throw a `PlatformNotSupported` btw to be more precise.

> I have the same question. If the else block is removed, what would happen on NetStandard? Should we be dropping NetStandard in TargetFrameworks tags as well?

Dropping `netstandard2.0` is always an option. But I think for tracing there is some good stuff that will still happen providing some value on older runtimes? Only suggesting we modify metrics here because it will literally do nothing with this shift.

Reply:
Fixed: b30c14cf1750e138c7256b21773927e03bcc16f5